### ISSUE TITLE
Use your code with a new data set

### ISSUE THREAD

**Issue by JohnTaylor2000** [amazon-science/earth-forecasting-transformer#45]
Could you run through the steps needed to use a different data set with your code please?

- input data format/layout
- can you use netcdf input files
- parameters that set the data file and data format characteristics
- how do you handle normalisation or standardisation of input data
- any other issues that need to be addressed when using a new data set

**Comment by gaozhihan** [comment]
Thanks for your interest. The most convenient way to use your own data is to implement a `pl.LightningDataModule` and integrate it into your training script. I will take the training script on MovingMNIST as example to illustrate how to do it.

- Implement your own `pl.LightningDataModule` ([example link](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/src/earthformer/datasets/moving_mnist/moving_mnist.py#L209)). You may refer to its [official doc](https://pytorch-lightning.readthedocs.io/en/1.7.4/data/datamodule.html) for more detailed instructions.
- Integrate your new `pl.LightningDataModule` into your training script ([example link](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/moving_mnist/train_cuboid_mnist.py#L431-L434)).
- Modify the data layout in the `forward()` method of the training `LightningModule` ([example link](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/moving_mnist/train_cuboid_mnist.py#L437)).

I address your questions in the following:

- Make sure the [input tensor](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/scripts/cuboid_transformer/moving_mnist/train_cuboid_mnist.py#L438) into the Earthformer has layout `"NTHWC"`. 
- Sure, as long as you implement a [PyTorch Dataset](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html) to load it.
- Please try to wrap all the details of loading the data into your `pl.LightningDataModule`.
- Typically we rescale the values to [0, 1]. ([example link](https://github.com/amazon-science/earth-forecasting-transformer/blob/e60ff41c7ad806277edc2a14a7a9f45585997bd7/src/earthformer/datasets/moving_mnist/moving_mnist.py#L99-L100))

**Comment by gaozhihan** [comment]
Let me close the issue for now. Please feel free to reopen if you find any problems.
